app/models/concerns/orchestration/tftp.rb

```diff
@@ -36,7 +36,9 @@ def tftp6
   def rebuild_tftp
     if tftp? || tftp6?
       begin
-        setTFTP
+        host.operatingsystem.template_kinds.each do |kind|
+          setTFTP(kind)
+        end
       rescue => e
```

Comment:
Can you move this _inside_ the template_kinds loop? Have a look at Orchestration::DNS for a similar example, where the rebuild method always tries to rebuild as much as possible, even if one fails, but then returns `false` on failure.